### PR TITLE
consumer for token staking service

### PR DESCRIPTION
**Pull request by prashantramangupta**
consumer for token staking service.

### CODE REVIEW

File: common/ipfs_util.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import ipfsapi
```
Comment (ksridharbabuus) [inline common/ipfs_util.py]: Why do we need IPFS for Staking Consumer?

File: staking/application/handlers/consumer_handlers.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+logger = get_logger(__name__)
+
+stake_window_repo = StakeWindowRepository()
+stake_holder_repo = StakeHolderRepository()
+
+
+@exception_handler(SLACK_HOOK=SLACK_HOOK, NETWORK_ID=NETWORK_ID, logger=logger)
+def open_for_stake_consumer_handler(event, context):
+    logger.info(f"Got OpenForStake Event {event}")
+    OpenForStakeEventConsumer(NETWORK['ws_provider'], IPFS_URL['url'], IPFS_URL['port'], NETWORK_ID).on_event(
```
Comment (ksridharbabuus) [inline staking/application/handlers/consumer_handlers.py]: Why IPFS URL for Stake?

File: staking/application/handlers/consumer_handlers.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        event)
+
+    return generate_lambda_response(200, StatusCode.OK)
+
+
+@exception_handler(SLACK_HOOK=SLACK_HOOK, NETWORK_ID=NETWORK_ID, logger=logger)
+def submit_stake_consumer_handler(event, context):
+    logger.info(f"Got OpenForStake Event {event}")
+    SubmitStakeEventConsumer(NETWORK['ws_provider'], IPFS_URL['url'], IPFS_URL['port'], NETWORK_ID).on_event(
```
Comment (ksridharbabuus) [inline staking/application/handlers/consumer_handlers.py]: Is the IPFS needed for the common code?

File: staking/application/handlers/consumer_handlers.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return generate_lambda_response(200, StatusCode.OK)
+
+
+@exception_handler(SLACK_HOOK=SLACK_HOOK, NETWORK_ID=NETWORK_ID, logger=logger)
+def withdraw_stake_consumer_handler(event, context):
+    logger.info(f"Got OpenForStake Event {event}")
+    WithdrawStakeEventConsumer(NETWORK['ws_provider'], IPFS_URL['url'], IPFS_URL['port'], NETWORK_ID).on_event(
+        event)
+
+    return generate_lambda_response(200, StatusCode.OK)
```
Comment (ksridharbabuus) [inline staking/application/handlers/consumer_handlers.py]: What about the following events?
AutoRenewStake
RenewStake
ClaimStake
UpdateAutoRenewal

File: staking/application/handlers/consumer_handlers.py (hunk 2 above)
Comment (prashantramangupta) [reply staking/application/handlers/consumer_handlers.py]: will remove if not needed.

**Pull request merged**